CRM/Core/DAO/permissions.php

```diff
@@ -177,6 +177,15 @@ function _civicrm_api3_permissions($entity, $action, &$params) {
     ),
   );
 
+  // Campaign permissions
+  $permissions['campaign'] = array(
+    'default' => array(
+      'access CiviCRM',
+      'manage campaign',
+    ),
```

Comment:
`default` means for all actions not specified. I think we should leave the default alone and target the actions we want to specify. I'd suggest:

    $permissions['campaign'] = array(
      'get' => array('access CiviCRM'),
      'create' => array(array('administer CiviCRM', 'administer CiviCampaign', 'manage campaign')),
      'update' => array(array('administer CiviCRM', 'administer CiviCampaign', 'manage campaign')),
      'delete' => array(array('administer CiviCRM', 'administer CiviCampaign')),
    );

Note the nested arrays translates to 'OR'. Some of these are iffy to me (like should 'manage campaign' give you permission to delete?) so maybe double-check the behavior in the rest of CiviCRM to verify